src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1383,8 +1572,13 @@ private function parse_next_tag() {
 					// Abruptly-closed empty comments are a sequence of dashes followed by `>`.
 					$span_of_dashes = strspn( $html, '-', $closer_at );
 					if ( '>' === $html[ $closer_at + $span_of_dashes ] ) {
-						$at = $closer_at + $span_of_dashes + 1;
-						continue;
+						// @todo This could go wrong if the closer is shorter than `<!---->` because there's no inside content.
```

Comment:
This todo should go even higher in this block. The document could end at `<!-->` or `<!--->` and we'd have a complete abruptly closed comment. It seems like the logic above is the problem:

https://github.com/WordPress/wordpress-develop/blob/103a556b6095fe61843719933db7ce2d0688ec7a/src/wp-includes/html-api/class-wp-html-tag-processor.php#L1564-L1570

It seems like there's a fix where we check to see if the comment starts with `>` or `->`. Then it would be a complete (abruptly closed) comment with no text content.

Reply:
I can clean up the `@todo`, but I think we're misunderstanding each other here. `<!-->` and `<!--->` are legitimate _invalid_ comments, but a problem arises with the introduction of modifiable text. in the case of `<!---->` we have a clear middle, between `<!--` and `-->` where we can insert text. With only two or three dashes we don't have the insertion point, so not only does this not have existing modifiable text, but we don't have a way to cleanly insert it without crossing a syntax boundary or rewriting the entire token.

This is mostly a modeling problem and not a safety one, because we can either rewrite it or refuse to modify its modifiable text, but I'm not happy so far with what I've explored as the mechanisms to indicate this special-case and how to update it.

Reply:
I rearranged the `@todo` but I left it in in 9f2992070ebbbb8970c96c115da2fb74a7ebfda4

this isn't necessary until we implement `set_modifiable_text()` so I wanted to leave a trace to remind ourselves in the future to not overlook this